src/libs/ascent/runtimes/flow_filters/ascent_runtime_vtkh_filters.cpp

```diff
@@ -3831,47 +3916,389 @@ VTKHParticleAdvection::execute()
+	}
+      }
+    }
+    else if(seed_type == "box")
     {
-      float x = seedBBox[0] + dx * distribution(generator);
-      float y = seedBBox[2] + dy * distribution(generator);
-      float z = seedBBox[4] + dz * distribution(generator);
-      seeds.push_back(vtkm::Particle({x,y,z}, i));
+      double dist_x, dist_y, dist_z;
+      double x_min, y_min, z_min;
+      double x_max, y_max, z_max;
+      if(n_seeds.has_child("extents_x"))
+      {
+        const Node &n_extents_x_vals = n_seeds["extents_x"];
+        const Node &n_extents_y_vals = n_seeds["extents_y"];
+        const Node &n_extents_z_vals = n_seeds["extents_z"];
+        Node n_extents_x_vals_dbls;
+        Node n_extents_y_vals_dbls;
+        Node n_extents_z_vals_dbls;
+        n_extents_x_vals.to_float64_array(n_extents_x_vals_dbls);
+        n_extents_y_vals.to_float64_array(n_extents_y_vals_dbls);
+        n_extents_z_vals.to_float64_array(n_extents_z_vals_dbls);
+        double* extents_x = n_extents_x_vals_dbls.as_double_ptr();
+        double* extents_y = n_extents_y_vals_dbls.as_double_ptr();
+        double* extents_z = n_extents_z_vals_dbls.as_double_ptr();
+	dist_x = extents_x[1] - extents_x[0];
+	dist_y = extents_y[1] - extents_y[0];
+	dist_z = extents_z[1] - extents_z[0];
+	x_min = extents_x[0];
+	y_min = extents_y[0];
+	z_min = extents_z[0];
+	x_max = extents_x[1];
+	y_max = extents_y[1];
+	z_max = extents_z[1];
+      }
+      else// whole dataset
+      {
+        vtkm::Bounds global_bounds = data.GetGlobalBounds();
+	dist_x = global_bounds.X.Length();
+	dist_y = global_bounds.Y.Length();
+	dist_z = global_bounds.Z.Length();
+	x_min = global_bounds.X.Min;
+	y_min = global_bounds.Y.Min;
+	z_min = global_bounds.Z.Min;
+	x_max = global_bounds.X.Max;
+	y_max = global_bounds.Y.Max;
+	z_max = global_bounds.Z.Max;
+      }
+      std::string sampling_type = n_seeds["sampling_type"].as_string();
+      std::string sampling_space = n_seeds["sampling_space"].as_string();
+      if(sampling_type != "uniform" && sampling_type != "random")
+      {
+        ASCENT_ERROR("Particle Advection box seeds accepts either 'uniform' or 'random' as the 'sampling_type'");
+      }
+
+
+      if(sampling_space == "interior")
+      {
+        if(sampling_type == "uniform")
+        {
+	  
+	  int num_seeds_x = n_seeds["num_seeds_x"].as_int();
+	  int num_seeds_y = n_seeds["num_seeds_y"].as_int();
+	  int num_seeds_z = n_seeds["num_seeds_z"].as_int();
+
+	  double dx = 1, dy = 1, dz = 1;
+	  if(num_seeds_x != 0)
+	    if(num_seeds_x != 1)
+              dx = dist_x/(num_seeds_x-1);
+	    else
+              dx = dist_x/num_seeds_x;
+	  if(num_seeds_y != 0)
+	    if(num_seeds_y != 1)
+              dy = dist_y/(num_seeds_y-1);
+	    else
+              dy = dist_y/num_seeds_y;
+	  if(num_seeds_z != 0)
+	    if(num_seeds_z != 1)
+              dz = dist_z/(num_seeds_z-1);
+	    else
+              dz = dist_z/num_seeds_z;
+ 
+	  double epsilon = 0.1;
```

Comment:
I was struggling to render seeds that are directly on the boundary, so I've been offsetting all my points my epsilon. 

Note for later for picking a better epsilon: 
      vtkm::Float64 lx = coordBounds.X.Length();
      vtkm::Float64 ly = coordBounds.Y.Length();
      vtkm::Float64 lz = coordBounds.Z.Length();
      vtkm::Float64 mag = vtkm::Sqrt(lx * lx + ly * ly + lz * lz);
      // same as used in vtk ospray
      constexpr vtkm::Float64 heuristic = 1000.;
      radius = static_cast<vtkm::Float32>(mag / heuristic);
      m_tube_size = radius;

Reply:
So I understand: Is it that the box is exactly aligned a data set boundary causing numeric sadness?

If so, the user could adjust the box to not be aligned.   Box params being correct is something user can control.

Reply:
@cyrush 

Exactly, anything on the boundary needs to be moved slightly or else we lose the particles. 

I guess the check should be if they are on the boundary and then adjust rather than adjusting ALL points like I am. 

Here are boundary points that are not and are adjusted using epsilon, respectively. 
![box_boundary_uniform_0000](https://github.com/Alpine-DAV/ascent/assets/24232495/8eefdd38-c7df-4865-8098-d9b6429c3b8a)
![box_boundary_uniform_0001](https://github.com/Alpine-DAV/ascent/assets/24232495/c9b6a51a-ba64-4e4e-ab76-7d4e71933fd9)